lib/spack/spack/cmd/buildcache.py

```diff
@@ -367,6 +373,23 @@ def _make_pool() -> MaybePool:
         return NoPool()
 
 
+def _skip_no_redistribute_for_public(specs):
+    remaining_specs = list()
+    removed_specs = list()
+    for spec in specs:
+        if spec.package.redistribute_binary:
+            remaining_specs.append(spec)
+        else:
+            removed_specs.append(spec)
+    if removed_specs:
+        tty.debug(
+            "The following specs will not be added to the binary cache"
+            " because their package.py file has marked them as"
+            " 'redistribute_binary = False': {0}".format(", ".join(s.name for s in removed_specs))
```

Comment:
This shouldn't mention `redistribute_binary` anymore:

```suggestion
            f" because they cannot be redistributed: {', '.join(s.name for s in removed_specs)}\n"
            "You can use `--private` to include them."
```

Also consider using `colify()`  for the list here.